src/webgpu/api/operation/adapter/info.spec.ts

```diff
@@ -39,3 +39,42 @@ g.test('adapter_info')
       `adapterInfo.device should be a normalized identifier. But it's '${adapterInfo.device}'`
     );
   });
+
+g.test('same_object')
+  .desc(
+    `
+GPUAdapter.info provides the same object each time it's accessed`
+  )
+  .fn(async t => {
+    const gpu = getGPU(t.rec);
+    const adapter = await gpu.requestAdapter();
+    assert(adapter !== null);
+
+    const adapterInfo1 = adapter.info;
+    const adapterInfo2 = adapter.info;
+    t.expect(adapterInfo1 === adapterInfo2);
+  });
+
+g.test('device_matches_adapter')
+  .desc(
+    `
+Test that GPUDevice.adapterInfo matches GPUAdapter.info`
+  )
+  .fn(async t => {
+    const gpu = getGPU(t.rec);
+    const adapter = await gpu.requestAdapter();
+    assert(adapter !== null);
+
+    const device = await t.requestDeviceTracked(adapter);
+    assert(device !== null);
+
+    assert(device.adapterInfo instanceof GPUAdapterInfo);
+    assert(adapter.info instanceof GPUAdapterInfo);
+
+    for (const k of Object.keys(GPUAdapterInfo.prototype)) {
```

Comment:
Landed the spec; didn't make adapter and device use the same object, so we should test that `adapter.info !== device.adapterInfo`.

Reply:
The spec doesn't say they have to be different object, so I wouldn't test that `adapter.info !== device.adapterInfo`. Note that Chromium experimental implementation actually returns the same object: https://chromium-review.googlesource.com/c/chromium/src/+/5980752/3/third_party/blink/renderer/modules/webgpu/gpu_device.cc

Reply:
It does implicitly require them to be different objects, because a new one is created the first time each of them is accessed, there's no opportunity to return the same one.

Reply:
I've updated this PR.
Do you know by any chance how to return a copy of `adapter_->info()` in Chromium?

Reply:
We should construct the `adapterInfo` when we construct `GPUDevice`, the same way we do for `info` when we construct `GPUAdapter`. The initialization will be the same so we should take this code, move it into a GPUAdapter method called something like `CreateAdapterInfoForAdapter()`, have it return the GPUAdapterInfo object, and then call it from both GPUAdapter::GPUAdapter and GPUDevice::GPUDevice.
https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/webgpu/gpu_adapter.cc;l=198-210;drc=0afc9ac9afcaab79fc54299039f4d27abf3a086d